crates/accelerate/src/two_qubit_decompose.rs

```diff
@@ -2344,6 +2344,352 @@ pub fn local_equivalence(weyl: PyReadonlyArray1<f64>) -> PyResult<[f64; 3]> {
+
+        if let Some(unitary_c1r) = unitary_c1r {
+            global_phase -= unitary_c1r.global_phase;
+            for gate in unitary_c1r.gates.into_iter() {
+                gates1.gates.push((Some(gate.0), gate.1, smallvec![0]));
+            }
+        }
+        if let Some(unitary_c1l) = unitary_c1l {
+            global_phase -= unitary_c1l.global_phase;
+            for gate in unitary_c1l.gates.into_iter() {
+                gates1.gates.push((Some(gate.0), gate.1, smallvec![1]));
+            }
+        }
+
+        gates1.global_phase = global_phase;
+        Ok(gates1)
+    }
+
+    ///  Initialize the KAK decomposition.
+    ///  Args:
+    ///      rxx_equivalent_gate: Gate that is locally equivalent to an :class:`.RXXGate`:
+    ///      :math:`U \sim U_d(\alpha, 0, 0) \sim \text{Ctrl-U}` gate.
+    ///  Raises:
+    ///      QiskitError: If the gate is not locally equivalent to an :class:`.RXXGate`.
+    fn new_inner(rxx_equivalent_gate: StandardGate) -> PyResult<Self> {
+        let atol = DEFAULT_ATOL;
+        let mut scales = Vec::new();
+        let test_angles = [0.2, 0.3, PI2];
+
+        for test_angle in test_angles {
+            if rxx_equivalent_gate.num_params() != 1 {
+                return Err(QiskitError::new_err(
+                    "Equivalent gate needs to take exactly 1 angle parameter.",
+                ));
+            }
+            let mat = rxx_equivalent_gate
+                .matrix(&[Param::Float(test_angle)])
+                .unwrap();
+            let decomp =
+                TwoQubitWeylDecomposition::new_inner(mat.view(), Some(DEFAULT_FIDELITY), None)?;
+
+            let mat_rxx = StandardGate::RXXGate
+                .matrix(&[Param::Float(test_angle)])
+                .unwrap();
+            let decomposer_rxx = TwoQubitWeylDecomposition::new_inner(
+                mat_rxx.view(),
+                None,
+                Some(Specialization::ControlledEquiv),
+            )?;
+            let decomposer_equiv = TwoQubitWeylDecomposition::new_inner(
+                mat.view(),
+                None,
+                Some(Specialization::ControlledEquiv),
+            )?;
+            let scale_a = decomposer_rxx.a / decomposer_equiv.a;
+
+            if (decomp.a * 2.0 - test_angle / scale_a).abs() > atol {
+                return Err(QiskitError::new_err(format!(
+                    "The gate {}
+                  is not equivalent to an RXXGate.",
+                    rxx_equivalent_gate.name()
+                )));
+            }
+
+            scales.push(scale_a);
+        }
+
+        let scale = scales[0];
+
+        // Check that all three tested angles give the same scale
+        for scale_val in scales.clone().into_iter() {
+            if !abs_diff_eq!(scale_val, scale, epsilon = atol) {
+                return Err(QiskitError::new_err(
+                    "Inconsistent scaling parameters in check.",
+                ));
+            }
+        }
```

Comment:
You can remove the clone here:

```suggestion
        for scale_val in &scales {
            if !abs_diff_eq!(scale_val, &scale, epsilon = atol) {
                return Err(QiskitError::new_err(
                    "Inconsistent scaling parameters in check.",
                ));
            }
        }
```

Reply:
done in [59ef018](https://github.com/Qiskit/qiskit/pull/13139/commits/59ef018f29c107d5c7cadbfa2442e5c0e94a8742)